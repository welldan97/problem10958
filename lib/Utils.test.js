const Utils = require('./Utils');

describe('Utils', () => {
  describe('getPermutationWithDuplicates', () => {
    it('returns permutation by index', () => {
      const placesLength = 3;
      const elementsLength = 2;
      let value;
      value = Utils.getPermutationWithDuplicates(
        placesLength,
        elementsLength,
        0,
      );
      expect(Array.from(value)).toEqual([0, 0, 0]);
      value = Utils.getPermutationWithDuplicates(
        placesLength,
        elementsLength,
        1,
      );
      expect(Array.from(value)).toEqual([0, 0, 1]);
      value = Utils.getPermutationWithDuplicates(
        placesLength,
        elementsLength,
        6,
      );
      expect(Array.from(value)).toEqual([1, 1, 0]);
    });
  });

  describe('getPermutation', () => {
    it('returns permutation by index', () => {
      const placesLength = 4;
      const value = [];
      for (let i = 0; i < 24; i++)
        value.push(Array.from(Utils.getPermutation(placesLength, i)));
      expect(Array.from(value)).toMatchSnapshot();
    });
  });
});
