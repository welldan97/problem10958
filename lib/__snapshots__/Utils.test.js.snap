// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utils getPermutation returns permutation by index 1`] = `
Array [
  Array [
    0,
    1,
    2,
    3,
  ],
  Array [
    0,
    1,
    3,
    2,
  ],
  Array [
    0,
    2,
    1,
    3,
  ],
  Array [
    0,
    2,
    3,
    1,
  ],
  Array [
    0,
    3,
    2,
    1,
  ],
  Array [
    0,
    3,
    1,
    2,
  ],
  Array [
    1,
    0,
    2,
    3,
  ],
  Array [
    1,
    0,
    3,
    2,
  ],
  Array [
    1,
    2,
    0,
    3,
  ],
  Array [
    1,
    2,
    3,
    0,
  ],
  Array [
    1,
    3,
    2,
    0,
  ],
  Array [
    1,
    3,
    0,
    2,
  ],
  Array [
    2,
    1,
    0,
    3,
  ],
  Array [
    2,
    1,
    3,
    0,
  ],
  Array [
    2,
    0,
    1,
    3,
  ],
  Array [
    2,
    0,
    3,
    1,
  ],
  Array [
    2,
    3,
    0,
    1,
  ],
  Array [
    2,
    3,
    1,
    0,
  ],
  Array [
    3,
    1,
    2,
    0,
  ],
  Array [
    3,
    1,
    0,
    2,
  ],
  Array [
    3,
    2,
    1,
    0,
  ],
  Array [
    3,
    2,
    0,
    1,
  ],
  Array [
    3,
    0,
    2,
    1,
  ],
  Array [
    3,
    0,
    1,
    2,
  ],
]
`;
